Sources/ClientRuntime/Retries/DefaultRetryStrategy/RetryQuota.swift

```diff
@@ -0,0 +1,62 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Keeps the retry quota count for one partition ID.
+///
+/// Is shared across all requests with the same partition ID; typically this also correlates to one network connection.
+final actor RetryQuota {
```

Comment:
This is the type that keeps the "token quota" for one network endpoint.
Implemented as an actor so multiple simultaneous requests can safely access it.